WeWatch/WeWatch/Custom TabBar/TabBarViewModel.swift

```diff
@@ -0,0 +1,80 @@
+//
+//  TabBarViewModel.swift
+//  WeWatch
+//
+//  Created by Anton on 30/12/2024.
+//
+
+import Foundation
+import SwiftUI
+
+internal final class TabBarViewModel: ObservableObject {
+    
+    @Published internal var homeViewIsActive: Bool = true
+    @Published internal var searchViewIsActive: Bool = false
+    @Published internal var bookmarkViewIsActive: Bool = false
+    @Published internal var selectedTab: TabViewType = .homeView
+    
+    internal enum TabViewType {
+        
+        case bookmark
+        case searchView
+        case homeView
+        
+        internal func activeView() -> AnyView {
+            switch self {
+            case.bookmark:
+                return AnyView(BookmarkView())
+            case.searchView:
+                return AnyView(SearchView())
+            case.homeView:
+                return AnyView(HomeView())
+            }
+        }
+    }
+    
+    internal func selectTab(_ tab: TabViewType) {
```

Comment:
This function should use `@MainActor` because it needs to run on the main thread to handle UI updates. You can learn more about it here: [MainActor in Swift](https://www.hackingwithswift.com/quick-start/concurrency/how-to-use-mainactor-to-run-code-on-the-main-queue).